Backfill ts tests for actual implementations. #268

sonybinhle opened the pull request:
I prepared the type test files based on our functional test files, please give me feedback.

jcchavezs commented on packages/zipkin/index.d.ts:
```diff
@@ -239,7 +239,7 @@ declare namespace zipkin {
       readonly annotationType: string;
       key: string;
-      value: string;
+      value: boolean | string | number;
```
I don't think we should change this, annotations should be only strings as that is what zipkin supports.

jcchavezs replied:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

jcchavezs replied:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

jcchavezs replied:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

jcchavezs replied:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

adriancole replied:
I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript..

jcchavezs replied:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

sonybinhle replied:
@jcchavezs @adriancole I updated the PR so with BinaryAnnotation the constructor will only allow string value.

> I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript..
The tests failed is my tests on type-tests folder, I saw the constructor accept boolean|string|number so I wrote tests to verify those three cases. But now when we only accept string then I also modified the ts tests.

> Just one question: do we need `readonly annotationType: string;` in the last case?
It is mandatory I guess since BinaryAnnotation implement the IAnnotation interface, I tried to remove it but getting errors on compiling. And also, the actual implementation has annotationType property.

sonybinhle replied:
@jcchavezs @adriancole I updated the PR so with BinaryAnnotation the constructor will only allow string value.

> I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript.
The tests failed is my tests on type-tests folder, I saw the constructor accept boolean|string|number, so I wrote tests to verify those three cases. But now when we only accept string then I also modified the ts tests.

> Just one question: do we need `readonly annotationType: string;` in the last case?
It is mandatory I guess since BinaryAnnotation implements the IAnnotation interface, I tried to remove it but getting errors on compiling. And also, the actual implementation has annotationType property.

sonybinhle replied:
@jcchavezs @adriancole I updated the PR so with BinaryAnnotation the constructor will only allow string value.

> I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript.

The tests failed is my tests on type-tests folder, I saw the constructor accept boolean|string|number, so I wrote tests to verify those three cases. But now when we only accept string then I also modified the ts tests.

> Just one question: do we need `readonly annotationType: string;` in the last case?

It is mandatory I guess since BinaryAnnotation implements the IAnnotation interface, I tried to remove it but getting errors on compiling. And also, the actual implementation has annotationType property.

sonybinhle commented:
@jcchavezs @adriancole I updated the PR so with BinaryAnnotation the constructor will only allow string value.

> I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript.

The tests failed is my tests on type-tests folder, I saw the constructor accept boolean|string|number, so I wrote tests to verify those three cases. But now when we only accept string then I also modified the ts tests.

> Just one question: do we need `readonly annotationType: string;` in the last case?

It is mandatory I guess since BinaryAnnotation implements the IAnnotation interface, I tried to remove it but getting errors on compiling. And also, the actual implementation has annotationType property.

sonybinhle commented:
@jcchavezs @adriancole I updated the PR so with BinaryAnnotation the
constructor will only allow string value.

> I think we constrain the typescript api to string, if the underlying
javascript can still pass otherwise. I didn't realize that the typescript
will affect unit tests written in javascript.

The tests failed is my tests on type-tests folder, I saw the constructor
accept boolean|string|number, so I wrote tests to verify those three cases.
But now when we only accept string then I also modified the ts tests.

> Just one question: do we need `readonly annotationType: string;` in the
last case?

It is mandatory I guess since BinaryAnnotation implements the IAnnotation
interface, I tried to remove it but getting errors on compiling. And also,
the actual implementation has annotationType property.

On Mon, Oct 22, 2018 at 7:57 AM Adrian Cole <notifications@github.com>
wrote:

> *@adriancole* commented on this pull request.
> ------------------------------
>
> In packages/zipkin/index.d.ts
> <https://github.com/openzipkin/zipkin-js/pull/271#discussion_r226905021>:
>
> > @@ -239,7 +239,7 @@ declare namespace zipkin {
>        constructor(key: string, value: boolean | string | number);
>        readonly annotationType: string;
>        key: string;
> -      value: string;
> +      value: boolean | string | number;
>
> I think we constrain the typescript api to string, if the underlying
> javascript can still pass otherwise. I didn't realize that the typescript
> will affect unit tests written in javascript..
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/zipkin-js/pull/271#discussion_r226905021>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIVE62B31s2O_0znvt0v34OGNiiilQC0ks5unV5UgaJpZM4XyYyt>
> .
>

jcchavezs commented:
Awesome work @sonybinhle!
